core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1455,7 +1456,11 @@ private[spark] object Utils extends Logging {
     val effectiveEnd = math.min(length, end)
```

Comment:
One major problem here is that offsetBytes(seqOfFiles) is supposed to return a string of data spanning multiple files based on the byte offset range. It uses the size of the files to calculate which files it needs partially/fully read. All that calculation is messed up if the files are compressed as each of them are lesser in size than the actual data they have. I am not sure whether it will work in that case. For example, I think reading a file fully in line 1507 will be completely broken as it will read X bytes from a compressed file of length X which when uncompressed will have >> X bytes.


Please add tests to figure this out, and fix it if needed.